### DISCUSSION TITLE
Window doesn't clear if I use Query<&Window> in my draw function

### DISCUSSION THREAD

**Discussion by jcarvajal288** [bevyengine/bevy#14801] [Q&A] [answered]
I'm having a problem where the window isn't clearing properly before each frame.  I've set a clear color, and I draw a grid of sprites.  When I resize the window, though, the sprite grid smears like in the image below:

![Screenshot_20240817_143117](https://github.com/user-attachments/assets/2b5be596-a8e5-46e6-a1af-3598cedd1369)

Here is my code:

```
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .insert_resource(Images::default())
        .insert_resource(LevelMap::default())
        .add_systems(Startup, (load_images, setup, draw).chain())
        .add_systems(Update, draw)
        .run();
}

fn setup(mut commands: Commands, mut clear_color: ResMut<ClearColor>) {
    commands.spawn(Camera2dBundle::default());
    clear_color.0 = Color::hsl(211., 1., 0.8);
}

fn draw(commands: Commands, level_map: Res<LevelMap>, images: Res<Images>, windows: Query<&Window>) {
    let window_center = Vec2::new(windows.single().resolution.width() / 2., windows.single().resolution.height() / 2.);
    level_map.draw(commands, &images, window_center);
}
```

I've figured out that the problem is the `windows` parameter in my `draw` function.  If I remove that parameter, the screen clears as I expect and there is no smearing, but I need that parameter  to get the screen resolution so I can put the sprite grid at the top left like I want.  How should I approach this?

Here is a link to the full code (error is on branch window-bug): https://github.com/jcarvajal288/puzzle-dungeon/tree/window-bug

**Comment by DasLixou** [comment, marked as answer]
Entities are persistent. You spawn new sprites in the `draw` system, which is in the `Update` schedule, being called every frame. So you create more and more tiles every frame. Moving the `draw` function into the `Startup` schedule and having another system updating their position should be the way to go.

**Reply by jcarvajal288** [reply]
Oh! Ok, so I'm not painting sprites on the screen each frame like I am in some other engines.  Things are starting to make sense.  Thank you.
